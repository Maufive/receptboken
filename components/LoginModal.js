import React, { Component } from "react";
import Modal from "react-modal";
import { MessageConsumer } from "./providers/MessageProvider";
import Login from "./Login";
import { Button } from "./styles/Button";
import { LoginContainer, customStyles } from "./styles/Jumbotron";

Modal.setAppElement("#__next");

class LoginModal extends Component {
	state = {
		isModalOpen: false
	};

	componentDidMount = () => {
		this.setState({ isModalOpen: this.props.open });
	};

	openModal = () => {
		this.setState({ isModalOpen: true });
	};

	closeModal = () => {
		this.setState({ isModalOpen: false });
	};

	render() {
		return (
			<LoginContainer>
				{this.props.primaryButton ? (
					<Button fullWidth primary onClick={this.openModal}>
						Logga in
					</Button>
				) : (
					<Button fullWidth onClick={this.openModal}>
						Logga in
					</Button>
				)}
				<Modal
					isOpen={this.state.isModalOpen}
					onAfterOpen={this.afterOpenModal}
					onRequestClose={this.closeModal}
					style={customStyles}
					contentLabel="Login Modal"
				>
					<MessageConsumer>
						{({ setMessage }) => (
							<Login setUser={this.props.setUser} setMessage={setMessage} />
						)}
					</MessageConsumer>
				</Modal>
			</LoginContainer>
		);
	}
}

export default LoginModal;
