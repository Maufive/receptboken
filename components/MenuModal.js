import React, { Component } from "react";
import PropTypes from "prop-types";
import Modal from "react-modal";
import Link from "next/link";
import Router from "next/router";
import UsersBoldIcon from "../svg/usersBold.svg";
import HeartBoldIcon from "../svg/likeBold.svg";
import ListBoldIcon from "../svg/listBold.svg";
import LogoutIcon from "../svg/logout.svg";
import MenuIcon from "../svg/menu-4.svg";
import CloseIcon from "../svg/closeBold.svg";
import AvatarIcon from "../svg/defaultAvatar.svg";
import {
	UserItem,
	Option,
	MenuContainer,
	ImgStyles
} from "./styles/DropdownStyles";
import { LoginContainer } from "./styles/Jumbotron";

Modal.setAppElement("#__next");

const ModalStyles = {
	content: {
		top: "50%",
		left: "50%",
		right: "auto",
		bottom: "auto",
		marginRight: "-50%",
		transform: "translate(-50%, -50%)",
		background: "rgba(21, 189, 118, 0.8)",
		width: "100vw",
		height: "100vh",
		zIndex: 10
	}
};

class UserDropdown extends Component {
	state = {
		open: false,
		isModalOpen: false
	};

	openModal = () => {
		this.setState({ isModalOpen: true });
	};

	closeModal = () => {
		this.setState({ isModalOpen: false });
	};

	logout = e => {
		localStorage.removeItem("jwtToken");
		Router.push("/");
		location.reload();
	};

	render() {
		const { user } = this.props;
		return (
			<LoginContainer>
				<UserItem onClick={this.openModal}>
					<MenuIcon />
					{user.fname}
					<ImgStyles>
						{user.photo ? (
							<img src={user.photo} alt="profilbild" />
						) : (
							<AvatarIcon />
						)}
					</ImgStyles>
				</UserItem>
				<Modal
					isOpen={this.state.isModalOpen}
					onAfterOpen={this.afterOpenModal}
					onRequestClose={this.closeModal}
					contentLabel="Menu Modal"
					style={ModalStyles}
					closeTimeoutMS={800}
				>
					<MenuContainer>
						<div>
							<Option onClick={this.closeModal}>
								<a>
									<CloseIcon />
									Stäng
								</a>
							</Option>
						</div>
						<div>
							<Option onClick={this.closeModal}>
								<Link
									href={{
										pathname: "/profile",
										query: {
											id: user._id
										}
									}}
								>
									<a>
										<UsersBoldIcon />
										Min profil
									</a>
								</Link>
							</Option>
						</div>
						<div>
							<Option onClick={this.closeModal}>
								<Link href="/saved">
									<a>
										<HeartBoldIcon />
										Sparade Recept
									</a>
								</Link>
							</Option>
						</div>
						<div>
							<Option onClick={this.closeModal}>
								<Link
									href={{
										pathname: "/inkopslistor",
										query: {
											id: user._id
										}
									}}
								>
									<a>
										<ListBoldIcon />
										Inköpslistor
									</a>
								</Link>
							</Option>
						</div>
						<div>
							<Option onClick={this.logout}>
								<a>
									<LogoutIcon />
									Logga ut
								</a>
							</Option>
						</div>
					</MenuContainer>
				</Modal>
			</LoginContainer>
		);
	}
}

export default UserDropdown;

UserDropdown.propTypes = {
	user: PropTypes.object.isRequired
};
