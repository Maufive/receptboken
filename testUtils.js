export const fakeRecept = {
	_id: "1234",
	author: "5678436374re67fwuf76",
	created: "Idag",
	description: ["Steg1", "Steg2", "Steg3"],
	id: "1337",
	ingredients: [
		{
			input: "Lök",
			numberOfUnits: "2",
			units: "dl"
		}
	],
	photo: "asdf123.jpg",
	reviews: [{ score: 1 }],
	servings: "2",
	tags: ["Mat", "Gott"],
	timeRequired: "20 min",
	title: "Mitt jättegoda recept"
};

export const fakeUser = {
	_id: "1234",
	username: "fakeuser@gmail.com",
	password: "1234",
	hearts: [],
	fname: "Niklas",
	lname: "Albinsson",
	description: "Hej jag heter Niklas",
	photo: "http://fakephoto.jpg"
};

export const fakeList = {
	list: [
		{
			input: "Lök",
			numberOfUnits: "5",
			units: "dl"
		},
		{
			input: "Tomater",
			numberOfUnits: "10",
			units: "st"
		}
	]
};
