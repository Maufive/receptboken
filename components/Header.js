import React, { Component } from "react";
import Link from "next/link";
import styled from "styled-components";
import LoginModal from "./LoginModal";
import UserDropdown from "./UserDropdown";
import MenuModal from "./MenuModal";
import GroceriesIcon from "../svg/groceries2.svg";

const Logo = styled.h2`
	margin: 0;
	margin-block-end: 0;
	margin-block-start: 0;
	color: ${props => props.theme.offWhite};
	font-family: "Montserrat";
	font-weight: 700;
	letter-spacing: 1px;
	font-size: 2rem;
`;
const StyledHeader = styled.header`
	display: flex;
	align-items: center;
	justify-content: space-between;
	padding: 1rem 2rem;
	height: 6rem;
	background: rgb(21, 189, 118);
	background: linear-gradient(
		90deg,
		rgba(21, 189, 118, 1) 0%,
		rgba(154, 202, 60, 1) 100%
	);
	width: 100%;
	margin-bottom: 3rem;
	> div > a {
		display: flex;
		align-items: center;
		padding: 0;
		svg {
			height: 3rem;
			width: 3rem;
			margin-right: 0.5rem;
		}
	}
`;

const ButtonContainer = styled.div`
	width: 50%;
	display: flex;
	flex-direction: column;
	align-items: flex-end;
`;

class Header extends Component {
	render() {
		const { user } = this.props;
		return (
			<StyledHeader>
				<div
					style={{
						width: "50%"
					}}
				>
					<Link href="/">
						<a>
							<GroceriesIcon />
							<Logo>Receptboken</Logo>
						</a>
					</Link>
				</div>
				{/* <Nav /> */}
				<ButtonContainer>
					{user ? (
						// <UserDropdown user={user} />
						<MenuModal user={user} />
					) : (
						<LoginModal setUser={this.props.setUser} />
					)}
				</ButtonContainer>
			</StyledHeader>
		);
	}
}

export default Header;
