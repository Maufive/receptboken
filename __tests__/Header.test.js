import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Header from "../components/Header";
import LoginModal from "../components/LoginModal";
import MenuModal from "../components/MenuModal";
import { fakeUser } from "../testUtils";

describe("<Header />", () => {
	it("renders without user", () => {
		const wrapper = shallow(<Header />);
		expect(toJSON(wrapper)).toMatchSnapshot();
		expect(wrapper.containsMatchingElement(<LoginModal />)).toBe(true);
	});

	it("renders with a user", () => {
		const wrapper = shallow(<Header user={fakeUser} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
		expect(wrapper.containsMatchingElement(<MenuModal user={fakeUser} />)).toBe(
			true
		);
	});
});
