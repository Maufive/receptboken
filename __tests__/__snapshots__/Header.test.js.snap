// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders with a user 1`] = `
<HeaderStyles__StyledHeader>
  <div
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <Link
      href="/"
    >
      <a>
        <HeaderStyles__Logo>
          RECEPTBOKEN
        </HeaderStyles__Logo>
      </a>
    </Link>
  </div>
  <HeaderStyles__ButtonContainer>
    <UserDropdown
      user={
        Object {
          "_id": "1234",
          "description": "Hej jag heter Niklas",
          "fname": "Niklas",
          "hearts": Array [],
          "lname": "Albinsson",
          "password": "1234",
          "photo": "http://fakephoto.jpg",
          "username": "fakeuser@gmail.com",
        }
      }
    />
  </HeaderStyles__ButtonContainer>
</HeaderStyles__StyledHeader>
`;

exports[`<Header /> renders without user 1`] = `
<HeaderStyles__StyledHeader>
  <div
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <Link
      href="/"
    >
      <a>
        <HeaderStyles__Logo>
          RECEPTBOKEN
        </HeaderStyles__Logo>
      </a>
    </Link>
  </div>
  <HeaderStyles__ButtonContainer>
    <LoginModal />
  </HeaderStyles__ButtonContainer>
</HeaderStyles__StyledHeader>
`;
